### PR TITLE
Fix glossary links and restrict `mkdocs-autorefs` updates

### PR DESCRIPTION
**Pull request by llucax**

### CODE REVIEW

**Comment by llucax** [comment]
OK, pinning the dependency for now as it requires some work to fix proper.

**Comment by llucax** [comment]
To fix this properly I think we need:

* https://github.com/mkdocstrings/autorefs/issues/52

File: docs/user-guide/glossary.md [hunk 1]
```diff
@@ -188,8 +188,8 @@ microgrid are connected with each other.  Some of the ways in which the SDK uses
 the component graph are:
 
   - figure out how to calculate high level metrics like
-[`grid_power`][frequenz.sdk.timeseries.grid.Grid.power],
-[`consumer_power`][frequenz.sdk.timeseries.consumer.Consumer.power],
+[`grid.power`][frequenz.sdk.timeseries.grid.Grid.power],
+[`consumer.power`][frequenz.sdk.timeseries.consumer.Consumer.power],
```
Comment (shsms) [inline docs/user-guide/glossary.md]: That's how it is represented elsewhere: https://github.com/frequenz-floss/frequenz-sdk-python/blob/4fb5a8802fe5f26206c6408de1389e7de9246bca/src/frequenz/sdk/microgrid/__init__.py#L80

Shouldn't it be like this? 
```suggestion
[`grid().power`][frequenz.sdk.timeseries.grid.Grid.power],
[`consumer().power`][frequenz.sdk.timeseries.consumer.Consumer.power],
```

File: docs/user-guide/glossary.md [hunk 2]
```diff
@@ -408,7 +408,8 @@ flowing into the component instead, this will be zero.
 
 In [microgrid](#microgrid) context sometimes used as the remaining difference
 between the [net consumption](#net-consumption) and the [battery](#battery)
-power, i.e. what we define as [grid power](#grid_power).
+power, i.e. what we define as
+[`grid.power`][frequenz.sdk.timeseries.grid.Grid.power].
```
Comment (shsms) [inline docs/user-guide/glossary.md]: and here
```suggestion
[`grid().power`][frequenz.sdk.timeseries.grid.Grid.power].
```

**Review by shsms — COMMENTED** [review]
Also a commit message says `<= 1.1`, but the code change says `< 1.1`

**Comment by llucax** [comment]
Updated to fix all of it.

**Review by daniel-zullo-frequenz — APPROVED** [review]

**Review by shsms — APPROVED** [review]

**Pull request merged**